example/matchers/consumer/tests/Service/MatchersTest.php

```diff
@@ -113,6 +113,24 @@ public function testGetMatchers(): void
                     [$this->matcher->regex(null, 'car|bike|motorbike')]
                 ),
                 'url' => $this->matcher->url('http://localhost:8080/users/1234/posts/latest', '.*(\\/users\\/\\d+\\/posts\\/latest)$', false),
+                'matchAll' => $this->matcher->matchAll(
```

Comment:
Maybe I am having a dim moment, but if `like` expects all values to match, and can accept nested matchers, why do we need to expose a new matcher to the end user.

Reply:
Yes, that's an example of when you might like to use both.